.github/workflows/build-test-ci.yml

```diff
@@ -17,44 +17,60 @@ jobs:
       matrix:
         os: [ubuntu-latest]
         cc: [gcc, clang]
+        bb: [meson, muon]
+      fail-fast: false
     runs-on: ${{ matrix.os }}
     env:
       CC: ${{ matrix.cc }}
+      BB: ${{ matrix.bb }}
     steps:
     - name: Install dependencies
-      run: sudo apt-get update && sudo apt-get install -y python3-pyelftools
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y python3-pyelftools python3-pip \
+                                libcap-dev libseccomp-dev ninja-build \
+                                pkg-config
+        case "$BB" in
+          meson)
+            sudo pip3 install meson
+            ;;
+          muon)
+            wget https://muon.build/releases/muon-amd64-linux-static -O muon
+            chmod +x muon
+            sudo install -Dm755 muon /usr/local/bin/muon
+            ;;
+        esac
     - uses: actions/checkout@v2
-    # Hack up the man pages as installing xmlto is very expensive.
-    # We'll test this in the autotools builder instead.
-    - name: Hack man pages
-      run: echo man/*.docbook | sed s:docbook:1:g | xargs touch
-    - run: make
-    - run: make check
-    - run: make install DESTDIR="${PWD}/root/"
-    - run: make debug
-    - run: make check
+    - run: |
+        case "$BB" in
+          meson)
+            cmd=meson
+            ;;
+          muon)
+            export PKG_CONFIG_PATH="/usr/lib/$(uname -m)-linux-gnu/pkgconfig/"
+            cmd="muon setup"
```

Comment:
Meson also uses `meson setup`.

The difference is that Meson has the IMHO surprising behavior of defaulting to an implicit `setup` if it doesn't recognize the second word as a subcommand. I really want to deprecate that behavior someday -- it means we don't have the freedom to create new subcommands because someone might use the same name as their build directory. We already had to choose "compile" instead of the original name of a "build" subcommand, because people often do `meson build` to specify a builddir at `./build/`.

In fact, you do that too :p but you have -D options in between. (To cater to muon, because muon is strict about POSIX utility syntax guidelines and requires the option-arguments to come before operands.)

Reply:
Hm, so common syntax would be `X setup -Da=b build`?

+1 on removing this behavior though ;)

Reply:
Yes, that's the common syntax.